common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -549,11 +549,22 @@ def should_force_account_creation():
         return (current_provider and
                 (current_provider.skip_email_verification or current_provider.send_to_registration_first))
 
+    def is_provider_saml():
+        current_provider = provider.Registry.get_from_pipeline({'backend': current_partial.backend, 'kwargs': kwargs})
+        return current_provider and current_provider in list(provider.Registry.get_enabled_by_backend_name('tpa-saml'))
+
     if not user:
-        if user_exists(details or {}):
-            # User has not already authenticated and the details sent over from
-            # identity provider belong to an existing user.
-            return dispatch_to_login()
+        if is_provider_saml():
+            email = details.get('email') if details else None
+            if user_exists({'email': email}):
```

Comment:
seems like we can DRY this up by having the if condition determine what the `details` param should be, and then pass that in to this check for dispatching to login

Reply:
Done